### DISCUSSION TITLE
Asus Store blocked my ip address. Is the stock checker still valid?

### DISCUSSION THREAD

**Discussion by jiglesiasc** [jef/streetmerchant#883] [Help]
Hello everyone,

As soon as I started to use this script, Asus Store started to display the following message:

_Dear Customers，

ASUS is dedicated to provide the best products and greatest service to our users over the past few decades. In order to improve our website quality, please be informed that we are undergoing scheduled software and hardware update. During this period, ASUS store will be temporarily unavailable. We sincerely apologize for any inconvenience caused during this period. Thank you so much for your patience and continued support._

Then I checked Asus Store from my cellphone (over the 4G network so it has a different IP address) and the site works perfectly. The phone over WiFi network also displays the same message, so my wired WAN IP address was blocked, no doubt.

But the script seems to be still working, showing "out of stock" status when checking Asus. Is that valid? or what is going on? Shouldn't the script be showing a connection error message or something?

**Comment by riskynacho** [comment]
No where in that message does it indicate your IP has been blocked. The message can be activated based on your ip address (whois lookup database).

**Comment by jiglesiasc** [comment]
> 
> 
> No where in that message does it indicate your IP has been blocked. The message can be activated based on your ip address (whois lookup database).

But either way I can't access Asus Store using my wired IP address, I mean if I open the web browser and type the URL for Asus Store I get that fake "maintenance message", if my IP has been blocked or not I have no idea.... there is some kind of restriction over my IP address though, some kind of URL redirect, hence my doubt if the script can or can't be trusted

**Comment by jiglesiasc** [comment]
Well now I'm getting this message, so I'm definitely banned, lol:

Access Denied
timestamp: 2020-11-23 03:56:28
event ID: 160850549201323644486100574325545329511
client IP address: 201.189.31.80
URL: https://store.asus.com/us/item_track
Please take a screenshot and contact your administrator

The script still shows no connection issue on Asus website which makes me wonder, what's going on?

**Comment by yumotto** [comment]
> Well now I'm getting this message, so I'm definitely banned, lol:
> 
> Access Denied
> timestamp: 2020-11-23 03:56:28
> event ID: 160850549201323644486100574325545329511
> client IP address: 201.189.31.80
> URL: https://store.asus.com/us/item_track
> Please take a screenshot and contact your administrator
> 
> The script still shows no connection issue on Asus website which makes me wonder, what's going on?

I had the same exact scenario happen to me. Seems to me that a lot of fake hits were appearing on Asus's site. The ban is temporarily. It'll only last about a week or two.

**Comment by jef** [comment]
Yep, I've been banned from Asus for awhile. I don't know the rate limiting they give, but it's definitely pretty strict.

**Comment by jef** [comment]
> But the script seems to be still working, showing "out of stock" status when checking Asus. Is that valid? or what is going on? Shouldn't the script be showing a connection error message or something?

We need to add a back off label here, as this should be corrected.

**Comment by jiglesiasc** [comment]
Fortunately my ISP provides me dynamic IP address so I just restarted my router and now Asus store is working again on my wired network, I did set the PAGE_SLEEP_MIN to like 90000 milliseconds  but Asus still banned my IP address. So I did set the PAGE_SLEEP_MIN to 4 minutes and I haven't been banned. Holy cow Asus really want to stop bot scalpers, lol... good to know. In this generation the Asus TUF RTX 3080 is the holy grail: That card has no cap issues, better temps, better noise levels, and better priced at $699

EDIT: I spoke too soon. Asus banned me again. Any chance to change the consult time between different models? The script consult the 3 models (tuf, tuf oc and strix) in like 4 - 8 seconds, maybe that time is too short for Asus and that's why I'm being banned

**Comment by jiglesiasc** [comment]
So the issue is definitely that when consulting the 3 models on Asus Store the script consult the 3 links within just a few seconds apart. Asus won't tolerate that.

What I did, I have now 3 scripts running:

script1: Runs all stores that I want to check except Asus Store.
sript2: Run only Asus Store and check only the TUF model every 5 minutes.
script3: Run only Asus Store and checks only the TUF OC model every 5 minutes but this script started 3 minutes after the script 2 so script2 and script3 won't overlap when consulting links on Asus Store website.

I have this "solution" running for more than 1 hour and I've been banned from Asus Store again.
